### PR TITLE
Extended page about functional programming

### PR DESCRIPTION
**Pull request by DCFargo**
Thought the topic was super interesting, also added contribution instructions for the Linux terminal. (Asssssummming it works on Windows but I haven't tested)

Thanks! :-)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -117,6 +117,46 @@ in Terminal, which should start the server (`Express server started on port 8080
 
 Then, simply open the page http://localhost:8080 to browse your own version of the website. Make any changes you would like (you don't have to rerun `node .`) in the IDE, and reload the page you are editing to see the changes. This means you can ensure you won't have any wacky formatting issues when you do a pull request - everything works exactly like it would on the actual, live website.
 
+### Setup on Linux Terminal
+Before starting it's important to install dependencies, the following uses `pacman` but a distribution-specific package manager should be supplemented.
```
Comment (Stinggyray) [inline README.md:121]: maybe a "such as apt"

File: resources/lambdas.md [hunk 2]
```diff
@@ -78,23 +78,64 @@ For more on lambda expressions, refer to the
...
 
-## What is Functional Programming
+## What is a Functional Programming Language?
 
-Functional programming is a programming paradigm where functions are "treated as first class citizens". Essentially, functions are treated as objects and can do
-anything that an object can (be passed as a parameter, returned, etc).
-The main idea behind functional programming is that it's programming without any or with minimal [side effects](https://en.wikipedia.org/wiki/Side_effect_(computer_science)). 
-In other words, it's programming where only local variables that are created inside the function can be modified. This means that no global variables, variables passed by reference, or parameterized variables can be
-modified for it to count as functional programming. An example would be:
+In general programming theory, there exists two main types of languages:
```
Comment (Stinggyray) [inline resources/lambdas.md:83]: It's a bit more nuanced than this (as you touch on further downwards). Many languages have a ton of functional features while being imperative in nature. They support different paradigms - it's not a black and white case of "this language is either imperative or functional" (like this implies) and more a case of how your code specifically is written.

Maybe specify that these are paradigms, and can be used in conjunction (to make it more clear).

File: resources/lambdas.md [hunk 3]
```diff
@@ -78,23 +78,64 @@ For more on lambda expressions, refer to the
...
+
+- [Imperative programming](https://en.wikipedia.org/wiki/Imperative_programming)
+- [Functional programming](https://en.wikipedia.org/wiki/Functional_programming)
+
+While these two will turn out to seem super different, they are [proven](https://en.wikipedia.org/wiki/Church%E2%80%93Turing_thesis) to be identical.
+
+In a nutshell, functional programming languages treat EVERYTHING as a function, including the basic definitions of `true` and `false`!
+
+From there, every series of actions is not defined as **what something should do**, but rather **what something is** as a recursive set of functions, then asking for a thing.
```
Comment (Stinggyray) [inline resources/lambdas.md:92]: These few lines of explanations above are not very intuitive.

Reading this line specifically, I'm still not entirely sure what it means. Maybe reword for clarity?

File: resources/lambdas.md [hunk 4]
```diff
@@ -78,23 +78,64 @@ For more on lambda expressions, refer to the
...
+```text
+f(x) = y
+
+-- snip --
+
+f(x) -> [what happens in f(x) stays in f(x)...] -> y
+```
+
+From these implementations, two things are realized; namely:
+- Working with sets becomes very useful
```
Comment (Stinggyray) [inline resources/lambdas.md:105]: How? Why?

File: resources/lambdas.md [hunk 5]
```diff
@@ -78,23 +78,64 @@ For more on lambda expressions, refer to the
...
 int[] otherThings = Arrays.stream(things).filter(i->i>5).toArray();
 ```
+
+There is a lot to note here about the aforementioned "side effects" as well.
+
 This program created a new array that only contained the integers that were greater than 5 from `things[]`. The important detail is that `things[]` was never altered; there were no side effects. 
 As a result, streams are an example of functional programing (but not the only one) since they never alter the original collection, and lambdas are too (as long as the input never gets altered). If you like to think of it mathematically, f(x) 
 must always return the same number (let's say a); but if you change the value of x it won't return a. If x never gets changed, then we have functional
-programming; but if it doesn't then we don't.
+programming; but if it doesn't, it must be pulling from an outside source, no bueno.
```
Comment (Stinggyray) [inline resources/lambdas.md:136]: It's not necessarily pulling from an outside source; it just got modified due to a "side effect".

File: resources/lambdas.md [hunk 6]
```diff
@@ -78,23 +78,64 @@ For more on lambda expressions, refer to the
...
+Functional programming languages are still widely used, particularly in data science for large processing operations. The YouTube channel Computerphile has a ton of 
+really neat videos on them if you're interested.
+
+## What does this have to do with what I'M working with?
+
+Great question. Well, since around 1993, a lot of imperative language developers started to realize how powerful functional programming can be.
+As you'd expect, Java is one of these functional-capable languages, and you may have even seen some of its functional features already. The most commonly used ones are:
+
+- Anonymous functions (lambdas): functions with no name
+- `map` function: applies a given function to every element in a collection
```
Comment (Stinggyray) [inline resources/lambdas.md:120]: You might want to specify these functions are called "higher-level" functions, since they are called with other functions as the parameters

File: resources/lambdas.md [hunk 7]
```diff
@@ -78,23 +78,64 @@ For more on lambda expressions, refer to the
...
+- `fold` function: collapses a collection into a single value according to a passed function
+- `filter` function: returns a collection with elements kept/removed according to a boolean function passed
+- Closures: while java does not have support for closures, these can be emulated with anonymous classes
```
Comment (Stinggyray) [inline resources/lambdas.md:123]: What is a closure?

File: resources/lambdas.md [hunk 8]
```diff
@@ -78,23 +78,64 @@ For more on lambda expressions, refer to the
...
+
+## What does this have to do with what I'M working with?
+
+Great question. Well, since around 1993, a lot of imperative language developers started to realize how powerful functional programming can be.
+As you'd expect, Java is one of these functional-capable languages, and you may have even seen some of its functional features already. The most commonly used ones are:
+
+- Anonymous functions (lambdas): functions with no name
+- `map` function: applies a given function to every element in a collection
+- `fold` function: collapses a collection into a single value according to a passed function
+- `filter` function: returns a collection with elements kept/removed according to a boolean function passed
```
Comment (Stinggyray) [inline resources/lambdas.md:122]: Usually called a "predicate"

File: resources/lambdas.md [hunk 9]
```diff
@@ -78,23 +78,64 @@ For more on lambda expressions, refer to the
...
+really neat videos on them if you're interested.
+
+## What does this have to do with what I'M working with?
+
+Great question. Well, since around 1993, a lot of imperative language developers started to realize how powerful functional programming can be.
+As you'd expect, Java is one of these functional-capable languages, and you may have even seen some of its functional features already. The most commonly used ones are:
+
+- Anonymous functions (lambdas): functions with no name
+- `map` function: applies a given function to every element in a collection
+- `fold` function: collapses a collection into a single value according to a passed function
```
Comment (Stinggyray) [inline resources/lambdas.md:121]: Does this actually exist in Java?
I know in JavaScript it's called reduce. You should probably also include that.

**Review by Stinggyray — CHANGES_REQUESTED** [review]
Some stuff I would maybe change.

**Comment by DCFargo** [comment]
Patched according to review. 😄 

Simplified, then expanded upon a couple bits, really hope I didn't open a can of worms lol

File: resources/lambdas.md (hunk 7 above)
Comment (DCFargo) [reply resources/lambdas.md]: Decided it wasn't important enough to include for the purposes of an intro, swapped out 👍

File: resources/lambdas.md (hunk 8 above)
Comment (DCFargo) [reply resources/lambdas.md]: Since it's mentioned above I just simplified to `lambdas`

File: resources/lambdas.md (hunk 3 above)
Comment (DCFargo) [reply resources/lambdas.md]: Reworded in terms of "problem solving" and "languages" to show how they interact.
Hopefully it makes it more clear.

**Review by Stinggyray — APPROVED** [review]
